sources/packages/backend/libs/services/src/system-users/system-users.service.ts

```diff
@@ -41,21 +35,32 @@ export class SystemUsersService {
     user.userName = SYSTEM_USER_USER_NAME;
     user.lastName = SYSTEM_USER_LAST_NAME;
     user.email = SERVICE_ACCOUNT_DEFAULT_USER_EMAIL;
-    return this.userRepo.save(user);
+    await this.userRepo
+      .createQueryBuilder()
+      .insert()
+      .into(User)
+      .values(user)
+      .orIgnore("ON CONSTRAINT users_user_name_key DO NOTHING")
+      .execute();
+
+    return this.getSystemUser();
   }
 
   /**
-   * Get system user details from the cache.
-   * If not found will create a new system user
-   * and return the user details.
-   * @return system user details.
+   * Load system user.
    */
-  async systemUser(): Promise<User> {
-    if (this.systemUserDetails) {
-      return this.systemUserDetails;
-    }
+  async loadSystemUser(): Promise<void> {
+    this.systemUser = await this.getOrCreateSystemUser();
```

Comment:
If this is the only place calling `getOrCreateSystemUser` do we need two methods?
Can `loadSystemUser` do exactly what `getOrCreateSystemUser` is doing?